sysml.library/Domain Libraries/Analysis/StateSpaceRepresentation.sysml

```diff
@@ -1,47 +1,51 @@
 package StateSpaceRepresentation {
     import ISQ::DurationValue;
+    import Quantities::VectorQuantityValue;
+    import VectorFunctions::*;
+
+    abstract attribute def StateSpace :> VectorQuantityValue;
+    abstract attribute def Input :> VectorQuantityValue;
+    abstract attribute def Output :> VectorQuantityValue;
 
-    // TODO: StateSpace should be a vector (rank-one tensor)
-    abstract attribute def StateSpace;
-    abstract attribute def Input;
-    abstract attribute def Output;
-   
     abstract calc def NextState(input: Input, stateSpace: StateSpace, timeStep: DurationValue): StateSpace;
     abstract calc def GetOutput(input: Input, stateSpace: StateSpace): Output;
-    
+
     /** Events to be received. */
-    abstract attribute def StateSpaceEvent;
-    attribute def ZeroCrossingEvent :> StateSpaceEvent;
-    
+    abstract action def StateSpaceEventDef;
+    action def ZeroCrossingEventDef :> StateSpaceEvent;
+
     /** Item for SSR connection */
     item def StateSpaceItem;
-    
+
     /** StateSpaceDynamics is the simplest form of State Space Representation,
         and nextState directly computes the stateSpace of the next timestep. */
     abstract action def StateSpaceDynamics {
-    	in attribute input: Input;
-    	
+        in attribute input: Input;
+
         abstract calc nextState: NextState;
         abstract calc getOutput: GetOutput;
         attribute stateSpace: StateSpace;
 
-       	out attribute output: Output = getOutput(input, stateSpace);
+        out attribute output: Output = getOutput(input, stateSpace);
     }
-   
+
     /** The definition of the time derivative of StateSpace, which means dx/dt, where x is StateSpace */
-    abstract attribute def StateDerivative;
-   
+    abstract attribute def StateDerivative :> VectorQuantityValue {
+        ref stateSpace: StateSpace;
+        constraint { stateSpace::order == order }
```

Comment:
I wonder where the right operand `order` resolves to?

Reply:
`order` would be `StateDerivative::order`.  This constrains the order of the derivative state space must be equal to the original state space.

Reply:
Ah yes, clear.